<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>S4 classes • BasicR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="S4 classes">
<meta property="og:description" content="BasicR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BasicR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/R_data_types.html">R Data types</a>
    </li>
    <li>
      <a href="../articles/S4_classes.html">S4 classes</a>
    </li>
    <li>
      <a href="../articles/S4_classes_exercises.html">S4 classes, exercises</a>
    </li>
    <li>
      <a href="../articles/basic_functions.html">Basic functions</a>
    </li>
    <li>
      <a href="../articles/control_structures.html">Control structures</a>
    </li>
    <li>
      <a href="../articles/data_types_exercises.html">Data types, exercises</a>
    </li>
    <li>
      <a href="../articles/data_types_exercises_solutions.html">Data types, solutions</a>
    </li>
    <li>
      <a href="../articles/exercises_control_exercises.html">Control structures, exercises</a>
    </li>
    <li>
      <a href="../articles/exercises_control_solutions.html">Control structures, solutions</a>
    </li>
    <li>
      <a href="../articles/functions_exercises.html">Functions, exercises</a>
    </li>
    <li>
      <a href="../articles/functions_exercises_solutions.html">Functions, solutions</a>
    </li>
    <li>
      <a href="../articles/intro.html">Course introduction</a>
    </li>
    <li>
      <a href="../articles/r_functions.html">Functions</a>
    </li>
    <li>
      <a href="../articles/references.html">References</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/CompEpigen/BasicR">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="S4_classes_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>S4 classes</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/CompEpigen/BasicR/blob/master/vignettes/S4_classes.Rmd"><code>vignettes/S4_classes.Rmd</code></a></small>
      <div class="hidden name"><code>S4_classes.Rmd</code></div>

    </div>

    
    
<p>S4 classes are the perfect examples of the OO programming: - stricter structure - they have representations: a list of slots or arguments by name and class - the slots have to have the correct type - methods - they also inherit the methods from the classes of the arguments, but inheritence is between S4 classes as well. - has to be created with the new constructor - they have accessor methods as well.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">setClass</span><span class="op">(</span><span class="st">"Person"</span>, <span class="fu">representation</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"character"</span>, age <span class="op">=</span> <span class="st">"numeric"</span><span class="op">)</span><span class="op">)</span>
<span class="kw">setClass</span><span class="op">(</span><span class="st">"Employee"</span>, <span class="fu">representation</span><span class="op">(</span>boss <span class="op">=</span> <span class="st">"Person"</span><span class="op">)</span>, contains <span class="op">=</span> <span class="st">"Person"</span><span class="op">)</span>

<span class="va">person1</span> <span class="op">&lt;-</span> <span class="fu">new</span><span class="op">(</span><span class="st">"Person"</span>, name<span class="op">=</span><span class="st">"Peter"</span>, age<span class="op">=</span><span class="fl">34</span><span class="op">)</span>
<span class="va">boss1</span> <span class="op">&lt;-</span> <span class="fu">new</span><span class="op">(</span><span class="st">"Employee"</span>, boss<span class="op">=</span><span class="va">person1</span>, name<span class="op">=</span><span class="st">"Lisa"</span>, age<span class="op">=</span><span class="fl">36</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">person1</span></code></pre></div>
<pre><code>## An object of class "Person"
## Slot "name":
## [1] "Peter"
## 
## Slot "age":
## [1] 34</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">boss1</span></code></pre></div>
<pre><code>## An object of class "Employee"
## Slot "boss":
## An object of class "Person"
## Slot "name":
## [1] "Peter"
## 
## Slot "age":
## [1] 34
## 
## 
## Slot "name":
## [1] "Lisa"
## 
## Slot "age":
## [1] 36</code></pre>
<p>Example class: SummarizedExperiment</p>
<p>Representation:</p>
<p><code>RangedSummarizedExperiment</code> is an extension of <code>SummarizedExperiment</code>. It uses <code>GRanges</code> to define the genomic regions.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span></code></pre></div>
<pre><code>## Loading required package: MatrixGenerics</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: 'MatrixGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:matrixStats':
## 
##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
##     colWeightedMeans, colWeightedMedians, colWeightedSds,
##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
##     rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     expand.grid, I, unname</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
<pre><code>## 
## Attaching package: 'Biobase'</code></pre>
<pre><code>## The following object is masked from 'package:MatrixGenerics':
## 
##     rowMedians</code></pre>
<pre><code>## The following objects are masked from 'package:matrixStats':
## 
##     anyMissing, rowMedians</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"airway_small"</span>,  package<span class="op">=</span><span class="st">"BasicR"</span><span class="op">)</span>

<span class="va">se</span> <span class="op">&lt;-</span> <span class="va">airway_small</span></code></pre></div>
<div id="accessors" class="section level2">
<h2 class="hasAnchor">
<a href="#accessors" class="anchor"></a>Accessors</h2>
<p>To know all this, look for the help and the vignette.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 8 rows and 9 columns
##            SampleName     cell      dex    albut        Run avgLength
##              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;
## SRR1039508 GSM1275862  N61311     untrt    untrt SRR1039508       126
## SRR1039509 GSM1275863  N61311     trt      untrt SRR1039509       126
## SRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126
## SRR1039513 GSM1275867  N052611    trt      untrt SRR1039513        87
## SRR1039516 GSM1275870  N080611    untrt    untrt SRR1039516       120
## SRR1039517 GSM1275871  N080611    trt      untrt SRR1039517       126
## SRR1039520 GSM1275874  N061011    untrt    untrt SRR1039520       101
## SRR1039521 GSM1275875  N061011    trt      untrt SRR1039521        98
##            Experiment    Sample    BioSample
##              &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;
## SRR1039508  SRX384345 SRS508568 SAMN02422669
## SRR1039509  SRX384346 SRS508567 SAMN02422675
## SRR1039512  SRX384349 SRS508571 SAMN02422678
## SRR1039513  SRX384350 SRS508572 SAMN02422670
## SRR1039516  SRX384353 SRS508575 SAMN02422682
## SRR1039517  SRX384354 SRS508576 SAMN02422673
## SRR1039520  SRX384357 SRS508579 SAMN02422683
## SRR1039521  SRX384358 SRS508580 SAMN02422677</code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 4001 rows and 0 columns</code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#assay(se)</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html">rowRanges</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></code></pre></div>
<pre><code>## GRangesList object of length 4001:
## $ENSG00000000003
## GRanges object with 17 ranges and 2 metadata columns:
##        seqnames            ranges strand |   exon_id       exon_name
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;     &lt;character&gt;
##    [1]        X 99883667-99884983      - |    667145 ENSE00001459322
##    [2]        X 99885756-99885863      - |    667146 ENSE00000868868
##    [3]        X 99887482-99887565      - |    667147 ENSE00000401072
##    [4]        X 99887538-99887565      - |    667148 ENSE00001849132
##    [5]        X 99888402-99888536      - |    667149 ENSE00003554016
##    ...      ...               ...    ... .       ...             ...
##   [13]        X 99890555-99890743      - |    667156 ENSE00003512331
##   [14]        X 99891188-99891686      - |    667158 ENSE00001886883
##   [15]        X 99891605-99891803      - |    667159 ENSE00001855382
##   [16]        X 99891790-99892101      - |    667160 ENSE00001863395
##   [17]        X 99894942-99894988      - |    667161 ENSE00001828996
##   -------
##   seqinfo: 722 sequences (1 circular) from an unspecified genome
## 
## ...
## &lt;4000 more elements&gt;</code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#assays(se)[[1]]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span></code></pre></div>
<pre><code>##                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516
## ENSG00000000003        679        448        873        408       1138
## ENSG00000000005          0          0          0          0          0
## ENSG00000000419        467        515        621        365        587
## ENSG00000000457        260        211        263        164        245
## ENSG00000000460         60         55         40         35         78
## ENSG00000000938          0          0          2          0          1
##                 SRR1039517 SRR1039520 SRR1039521
## ENSG00000000003       1047        770        572
## ENSG00000000005          0          0          0
## ENSG00000000419        799        417        508
## ENSG00000000457        331        233        229
## ENSG00000000460         63         76         60
## ENSG00000000938          0          0          0</code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">airway_small</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [[1]]
## Experiment data
##   Experimenter name: Himes BE 
##   Laboratory: NA 
##   Contact information:  
##   Title: RNA-Seq transcriptome profiling identifies CRISPLD2 as a glucocorticoid responsive gene that modulates cytokine function in airway smooth muscle cells. 
##   URL: http://www.ncbi.nlm.nih.gov/pubmed/24926665 
##   PMIDs: 24926665 
## 
##   Abstract: A 226 word abstract is available. Use 'abstract' method.</code></pre>
<p>You can access each data “natively”, and even change it, but you shouldn’t do that. Why? It is easy to screw it up…</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">se</span><span class="op">@</span><span class="va">rowRanges</span></code></pre></div>
<pre><code>## GRangesList object of length 4001:
## $ENSG00000000003
## GRanges object with 17 ranges and 2 metadata columns:
##        seqnames            ranges strand |   exon_id       exon_name
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;     &lt;character&gt;
##    [1]        X 99883667-99884983      - |    667145 ENSE00001459322
##    [2]        X 99885756-99885863      - |    667146 ENSE00000868868
##    [3]        X 99887482-99887565      - |    667147 ENSE00000401072
##    [4]        X 99887538-99887565      - |    667148 ENSE00001849132
##    [5]        X 99888402-99888536      - |    667149 ENSE00003554016
##    ...      ...               ...    ... .       ...             ...
##   [13]        X 99890555-99890743      - |    667156 ENSE00003512331
##   [14]        X 99891188-99891686      - |    667158 ENSE00001886883
##   [15]        X 99891605-99891803      - |    667159 ENSE00001855382
##   [16]        X 99891790-99892101      - |    667160 ENSE00001863395
##   [17]        X 99894942-99894988      - |    667161 ENSE00001828996
##   -------
##   seqinfo: 722 sequences (1 circular) from an unspecified genome
## 
## ...
## &lt;4000 more elements&gt;</code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">se</span><span class="op">@</span><span class="va">metadata</span></code></pre></div>
<pre><code>## [[1]]
## Experiment data
##   Experimenter name: Himes BE 
##   Laboratory: NA 
##   Contact information:  
##   Title: RNA-Seq transcriptome profiling identifies CRISPLD2 as a glucocorticoid responsive gene that modulates cytokine function in airway smooth muscle cells. 
##   URL: http://www.ncbi.nlm.nih.gov/pubmed/24926665 
##   PMIDs: 24926665 
## 
##   Abstract: A 226 word abstract is available. Use 'abstract' method.</code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">###don't do this:</span>
<span class="va">se</span><span class="op">@</span><span class="va">rowRanges</span> <span class="op">&lt;-</span> <span class="va">se</span><span class="op">@</span><span class="va">rowRanges</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3000</span><span class="op">]</span>
<span class="va">se</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 4001 8 
## metadata(1): ''
## assays(1): counts
## rownames(3000): ENSG00000000003 ENSG00000000005 ... ENSG00000162951
##   ENSG00000162959
## rowData names(0):
## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">se</span><span class="op">@</span><span class="va">rowRanges</span></code></pre></div>
<pre><code>## GRangesList object of length 3000:
## $ENSG00000000003
## GRanges object with 17 ranges and 2 metadata columns:
##        seqnames            ranges strand |   exon_id       exon_name
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;     &lt;character&gt;
##    [1]        X 99883667-99884983      - |    667145 ENSE00001459322
##    [2]        X 99885756-99885863      - |    667146 ENSE00000868868
##    [3]        X 99887482-99887565      - |    667147 ENSE00000401072
##    [4]        X 99887538-99887565      - |    667148 ENSE00001849132
##    [5]        X 99888402-99888536      - |    667149 ENSE00003554016
##    ...      ...               ...    ... .       ...             ...
##   [13]        X 99890555-99890743      - |    667156 ENSE00003512331
##   [14]        X 99891188-99891686      - |    667158 ENSE00001886883
##   [15]        X 99891605-99891803      - |    667159 ENSE00001855382
##   [16]        X 99891790-99892101      - |    667160 ENSE00001863395
##   [17]        X 99894942-99894988      - |    667161 ENSE00001828996
##   -------
##   seqinfo: 722 sequences (1 circular) from an unspecified genome
## 
## ...
## &lt;2999 more elements&gt;</code></pre>
</div>
<div id="methods" class="section level2">
<h2 class="hasAnchor">
<a href="#methods" class="anchor"></a>Methods</h2>
<div id="subsetting" class="section level3">
<h3 class="hasAnchor">
<a href="#subsetting" class="anchor"></a>subsetting</h3>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># subsetting is very easy, just like with a data.frame</span>

<span class="va">se</span> <span class="op">&lt;-</span> <span class="va">se</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2000</span>,<span class="op">]</span>

<span class="va">se</span> <span class="op">&lt;-</span> <span class="va">se</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>

<span class="va">se</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 2000 3 
## metadata(1): ''
## assays(1): counts
## rownames(2000): ENSG00000000003 ENSG00000000005 ... ENSG00000092758
##   ENSG00000092820
## rowData names(0):
## colnames(3): SRR1039508 SRR1039509 SRR1039512
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
</div>
<div id="combining" class="section level3">
<h3 class="hasAnchor">
<a href="#combining" class="anchor"></a>Combining</h3>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rowbinded</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">se</span>, <span class="va">se</span><span class="op">)</span>
<span class="va">rowbinded</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 4000 3 
## metadata(2): '' ''
## assays(1): counts
## rownames(4000): ENSG00000000003 ENSG00000000005 ... ENSG00000092758
##   ENSG00000092820
## rowData names(0):
## colnames(3): SRR1039508 SRR1039509 SRR1039512
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">colbinded</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">se</span>, <span class="va">se</span><span class="op">)</span>
<span class="va">colbinded</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 2000 6 
## metadata(2): '' ''
## assays(1): counts
## rownames(2000): ENSG00000000003 ENSG00000000005 ... ENSG00000092758
##   ENSG00000092820
## rowData names(0):
## colnames(6): SRR1039508 SRR1039509 ... SRR1039509 SRR1039512
## colData names(9): SampleName cell ... Sample BioSample</code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">roi</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span>seqnames<span class="op">=</span><span class="st">"X"</span>, ranges<span class="op">=</span><span class="fl">99800000</span><span class="op">:</span><span class="fl">99850000</span><span class="op">)</span>
<span class="va">se_1</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">se</span>, <span class="va">roi</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="granges-and-grangeslist" class="section level2">
<h2 class="hasAnchor">
<a href="#granges-and-grangeslist" class="anchor"></a>Granges and GRangesList</h2>
<p><code>GRanges</code> objects store information about Genomic ranges with chromosome, position and additional information. Offers a wide range of functionality. <code>GRangesList</code> is a “grouped” version of <code>GRnages</code>, it is similar to a normal list, but with additional functionality. Both of them are able to handle strand information as well.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></span><span class="op">)</span>

<span class="va">gr1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>
    seqnames <span class="op">=</span> <span class="st">"chr2"</span>,
    ranges <span class="op">=</span> <span class="fu">IRanges</span><span class="op">(</span><span class="fl">103</span>, <span class="fl">106</span><span class="op">)</span>,
    strand <span class="op">=</span> <span class="st">"+"</span>,
    score <span class="op">=</span> <span class="fl">5L</span>, GC <span class="op">=</span> <span class="fl">0.45</span><span class="op">)</span>
<span class="va">gr2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>
    seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr3"</span><span class="op">)</span>,
    ranges <span class="op">=</span> <span class="fu">IRanges</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">107</span>, <span class="fl">113</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>,
    strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"-"</span><span class="op">)</span>,
    score <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">4</span>, GC <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="va">gr3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span><span class="st">"chr2:102-107"</span><span class="op">)</span>

<span class="va">gr4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html">makeGRangesFromDataFrame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>chr<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span><span class="op">)</span>, 
                                           start<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">104</span>, <span class="fl">104</span><span class="op">)</span>, end<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">108</span>, <span class="fl">105</span><span class="op">)</span>,
                                           name<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gene1"</span>, <span class="st">"gene2"</span><span class="op">)</span><span class="op">)</span>, keep.extra.columns <span class="op">=</span> <span class="cn">T</span>, ignore.strand <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="va">grl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html">GRangesList</a></span><span class="op">(</span><span class="st">"txA"</span> <span class="op">=</span> <span class="va">gr1</span>, <span class="st">"txB"</span> <span class="op">=</span> <span class="va">gr2</span><span class="op">)</span>
<span class="va">grl</span></code></pre></div>
<pre><code>## GRangesList object of length 2:
## $txA
## GRanges object with 1 range and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr2   103-106      + |         5      0.45
##   -------
##   seqinfo: 3 sequences from an unspecified genome; no seqlengths
## 
## $txB
## GRanges object with 2 ranges and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr1   107-109      + |         3       0.3
##   [2]     chr3   113-115      - |         4       0.5
##   -------
##   seqinfo: 3 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#seqinfo(gr1) &lt;- Seqinfo(genome="hg38")</span>
<span class="co">#can have names for each row</span></code></pre></div>
<div id="accessors-1" class="section level3">
<h3 class="hasAnchor">
<a href="#accessors-1" class="anchor"></a>Accessors</h3>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">seqnames</span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code>## factor-Rle of length 1 with 1 run
##   Lengths:    1
##   Values : chr2
## Levels(1): chr2</code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">seqlevels</span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "chr2"</code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ranges</span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code>## IRanges object with 1 range and 0 metadata columns:
##           start       end     width
##       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
##   [1]       103       106         4</code></pre>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 103</code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 106</code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">strand</span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code>## factor-Rle of length 1 with 1 run
##   Lengths: 1
##   Values : +
## Levels(3): + - *</code></pre>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">width</span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 4</code></pre>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 1</code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">mcols</span><span class="op">(</span><span class="va">gr4</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 2 rows and 1 column
##          name
##   &lt;character&gt;
## 1       gene1
## 2       gene2</code></pre>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">mcols</span><span class="op">(</span><span class="va">gr4</span><span class="op">)</span><span class="op">$</span><span class="va">name</span></code></pre></div>
<pre><code>## [1] "gene1" "gene2"</code></pre>
</div>
<div id="subsetting-1" class="section level3">
<h3 class="hasAnchor">
<a href="#subsetting-1" class="anchor"></a>Subsetting</h3>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#data.frame-like</span>

<span class="va">gr1</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span></code></pre></div>
<pre><code>## GRanges object with 1 range and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr2   103-106      + |         5      0.45
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gr4</span><span class="op">[</span><span class="fl">1</span>, <span class="st">"name"</span><span class="op">]</span></code></pre></div>
<pre><code>## GRanges object with 1 range and 1 metadata column:
##       seqnames    ranges strand |        name
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
##   [1]     chr1   104-108      * |       gene1
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># subset function</span>
<span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="va">strand</span> <span class="op">==</span> <span class="st">"+"</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 1 range and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr2   103-106      + |         5      0.45
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="splitting-and-combining" class="section level3">
<h3 class="hasAnchor">
<a href="#splitting-and-combining" class="anchor"></a>Splitting and combining</h3>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span><span class="op">(</span><span class="va">gr2</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span><span class="op">(</span><span class="va">gr4</span>, <span class="op">~</span> <span class="va">name</span><span class="op">)</span></code></pre></div>
<pre><code>## GRangesList object of length 2:
## $gene1
## GRanges object with 1 range and 1 metadata column:
##       seqnames    ranges strand |        name
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
##   [1]     chr1   104-108      * |       gene1
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths
## 
## $gene2
## GRanges object with 1 range and 1 metadata column:
##       seqnames    ranges strand |        name
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
##   [1]     chr2   104-105      * |       gene2
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">sp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="va">sp</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr1   107-109      + |         3       0.3
##   [2]     chr3   113-115      - |         4       0.5
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/stack.html">stack</a></span><span class="op">(</span><span class="va">sp</span>, index.var<span class="op">=</span><span class="st">"name"</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 3 metadata columns:
##       seqnames    ranges strand |  name     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;Rle&gt; &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr1   107-109      + |     1         3       0.3
##   [2]     chr3   113-115      - |     2         4       0.5
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="aggregating" class="section level3">
<h3 class="hasAnchor">
<a href="#aggregating" class="anchor"></a>Aggregating</h3>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html">aggregate</a></span><span class="op">(</span><span class="va">gr2</span>, <span class="va">score</span> <span class="op">~</span> <span class="va">strand</span>, <span class="va">mean</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 2 rows and 2 columns
##     strand     score
##   &lt;factor&gt; &lt;numeric&gt;
## 1        +         3
## 2        -         4</code></pre>
</div>
<div id="interval-operations" class="section level3">
<h3 class="hasAnchor">
<a href="#interval-operations" class="anchor"></a>Interval operations</h3>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html">flank</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="fl">10</span>, both<span class="op">=</span><span class="cn">T</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 1 range and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr2    93-112      + |         5      0.45
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/intra-range-methods.html">shift</a></span><span class="op">(</span><span class="va">gr2</span>, <span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr1   112-114      + |         3       0.3
##   [2]     chr3   118-120      - |         4       0.5
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html">resize</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="fl">20</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 1 range and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr2   103-122      + |         5      0.45
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="va">gr3</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr2   103-106      + |         5      0.45
##   [2]     chr2   102-107      * |      &lt;NA&gt;        NA
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html">reduce</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="va">gr3</span><span class="op">)</span><span class="op">)</span> <span class="co"># combines the overlapping ranges</span></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr2   103-106      +
##   [2]     chr2   102-107      *
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html">reduce</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="va">gr3</span><span class="op">)</span>, ignore.strand<span class="op">=</span><span class="cn">T</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 1 range and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr2   102-107      *
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="comapring-2-granges" class="section level3">
<h3 class="hasAnchor">
<a href="#comapring-2-granges" class="anchor"></a>Comapring 2 GRanges</h3>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/setops-methods.html">union</a></span><span class="op">(</span><span class="va">gr2</span>, <span class="va">gr4</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in .Seqinfo.mergexy(x, y): Each of the 2 combined objects has sequence levels not in the other:
##   - in 'x': chr3
##   - in 'y': chr2
##   Make sure to always combine/compare objects based on the same reference
##   genome (use suppressWarnings() to suppress this warning).</code></pre>
<pre><code>## GRanges object with 4 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr1   107-109      +
##   [2]     chr1   104-108      *
##   [3]     chr3   113-115      -
##   [4]     chr2   104-105      *
##   -------
##   seqinfo: 3 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#intersect(gr2, gr4)</span>
<span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/setops-methods.html">setdiff</a></span><span class="op">(</span><span class="va">gr2</span>, <span class="va">gr4</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in .Seqinfo.mergexy(x, y): Each of the 2 combined objects has sequence levels not in the other:
##   - in 'x': chr3
##   - in 'y': chr2
##   Make sure to always combine/compare objects based on the same reference
##   genome (use suppressWarnings() to suppress this warning).</code></pre>
<pre><code>## GRanges object with 2 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr1   107-109      +
##   [2]     chr3   113-115      -
##   -------
##   seqinfo: 3 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html">findOverlaps</a></span><span class="op">(</span><span class="va">gr2</span>, <span class="va">gr4</span>, ignore.strand<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in .Seqinfo.mergexy(x, y): Each of the 2 combined objects has sequence levels not in the other:
##   - in 'x': chr3
##   - in 'y': chr2
##   Make sure to always combine/compare objects based on the same reference
##   genome (use suppressWarnings() to suppress this warning).</code></pre>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html">countOverlaps</a></span><span class="op">(</span><span class="va">gr2</span>, <span class="va">gr4</span>, ignore.strand<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in .Seqinfo.mergexy(x, y): Each of the 2 combined objects has sequence levels not in the other:
##   - in 'x': chr3
##   - in 'y': chr2
##   Make sure to always combine/compare objects based on the same reference
##   genome (use suppressWarnings() to suppress this warning).</code></pre>
<pre><code>## [1] 1 0</code></pre>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html">subsetByOverlaps</a></span><span class="op">(</span><span class="va">gr2</span>, <span class="va">gr4</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in .Seqinfo.mergexy(x, y): Each of the 2 combined objects has sequence levels not in the other:
##   - in 'x': chr3
##   - in 'y': chr2
##   Make sure to always combine/compare objects based on the same reference
##   genome (use suppressWarnings() to suppress this warning).</code></pre>
<pre><code>## GRanges object with 1 range and 2 metadata columns:
##       seqnames    ranges strand |     score        GC
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;
##   [1]     chr1   107-109      + |         3       0.3
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
<div id="deseq2" class="section level2">
<h2 class="hasAnchor">
<a href="#deseq2" class="anchor"></a>DeSeq2</h2>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"pasilla"</span><span class="op">)</span>
<span class="va">pasCts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,
                      <span class="st">"pasilla_gene_counts.tsv"</span>,
                      package<span class="op">=</span><span class="st">"pasilla"</span>, mustWork<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">pasAnno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,
                       <span class="st">"pasilla_sample_annotation.csv"</span>,
                       package<span class="op">=</span><span class="st">"pasilla"</span>, mustWork<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">cts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="va">pasCts</span>,sep<span class="op">=</span><span class="st">"\t"</span>,row.names<span class="op">=</span><span class="st">"gene_id"</span><span class="op">)</span><span class="op">)</span>
<span class="va">coldata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="va">pasAnno</span>, row.names<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="va">coldata</span> <span class="op">&lt;-</span> <span class="va">coldata</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"condition"</span>,<span class="st">"type"</span><span class="op">)</span><span class="op">]</span>

<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"fb"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span><span class="op">)</span>
<span class="va">cts</span> <span class="op">&lt;-</span> <span class="va">cts</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span><span class="op">]</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/mikelove/DESeq2">"DESeq2"</a></span><span class="op">)</span>
<span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span><span class="op">(</span>countData <span class="op">=</span> <span class="va">cts</span>,
                              colData <span class="op">=</span> <span class="va">coldata</span>,
                              design <span class="op">=</span> <span class="op">~</span> <span class="va">condition</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds</span></code></pre></div>
<pre><code>## class: DESeqDataSet 
## dim: 14599 7 
## metadata(1): version
## assays(1): counts
## rownames(14599): FBgn0000003 FBgn0000008 ... FBgn0261574 FBgn0261575
## rowData names(0):
## colnames(7): treated1 treated2 ... untreated3 untreated4
## colData names(2): condition type</code></pre>
<div class="sourceCode" id="cb122"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds</span> <span class="op">&lt;-</span> <span class="va">dds</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">200</span>,<span class="op">]</span>
<span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span>
<span class="va">res</span></code></pre></div>
<pre><code>## log2 fold change (MLE): condition untreated vs treated 
## Wald test p-value: condition untreated vs treated 
## DataFrame with 200 rows and 6 columns
##                baseMean log2FoldChange     lfcSE       stat      pvalue
##               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;   &lt;numeric&gt;
## FBgn0000003    0.172207     -0.9986796  3.806863 -0.2623366   0.7930619
## FBgn0000008   95.873798      0.0140036  0.219868  0.0636906   0.9492166
## FBgn0000014    1.090634      0.5233543  2.215680  0.2362048   0.8132737
## FBgn0000015    0.853818      1.8909340  2.149959  0.8795209   0.3791189
## FBgn0000017 4392.045082      0.2549107  0.116978  2.1791389   0.0293213
## ...                 ...            ...       ...        ...         ...
## FBgn0000567   263.99287       1.239187  0.166174   7.457181 8.83932e-14
## FBgn0000568  3655.58621       0.220186  0.210651   1.045263 2.95901e-01
## FBgn0000575  1435.99029      -0.300818  0.175862  -1.710531 8.71677e-02
## FBgn0000576     0.00000             NA        NA         NA          NA
## FBgn0000577     0.35367       1.822069  3.491008   0.521932 6.01718e-01
##                    padj
##               &lt;numeric&gt;
## FBgn0000003    0.994466
## FBgn0000008    0.996580
## FBgn0000014    0.994466
## FBgn0000015    0.817888
## FBgn0000017    0.189072
## ...                 ...
## FBgn0000567 8.26476e-12
## FBgn0000568 7.88180e-01
## FBgn0000575 3.97570e-01
## FBgn0000576          NA
## FBgn0000577 9.94466e-01</code></pre>
<p>```</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Reka Toth, Rajbir Nath Batra.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
